cloudinit/net/dhcp.py

```diff
@@ -149,6 +157,19 @@ def dhcp_discovery(dhclient_cmd_path, interface, dhcp_log_func=None):
         "-sf",
         "/bin/true",
     ]
+    is_ib_interface = read_sys_net_safe(interface, "type") == "32"
+    if is_ib_interface:
+        dhcp_client_identifier = "20:%s" % get_interface_mac(interface)[36:]
+        interface_dhclient_content = (
+            'interface "%s" '
+            "{send dhcp-client-identifier %s;}"
+            % (interface, dhcp_client_identifier)
+        )
+        tmp_dir = temp_utils.get_tmp_ancestor(needs_exe=True)
+        file_name = os.path.join(tmp_dir, interface + "-dhclient.conf")
```

Comment:
I'd really rather not go down the path of dealing with temp configs. Why not just use the flag?
```
-I <dhcp-client-identifier>
    Specify the dhcp-client-identifier option to send to the DHCP server.
```

Reply:
so using -I will be send string and not as hex as we would like.  we didn't find away to do it without file and it seem to be consistent with how NetowrkManager is doing it.

Reply:
Unfortunately it looks like `dhclient` changed the meaning of this flag at some point:

From the latest man page:
```
       -I     Use the standard DDNS scheme from RFCs 4701 & 4702.
```

I guess this means that your current approach is the most cross-platform one at the moment, even if that's less ideal than a commandline argument.